src/main/Main.hs

```diff
@@ -608,7 +632,7 @@ uploadCmd args go = do
 
 sdistCmd :: [String] -> GlobalOpts -> IO ()
 sdistCmd dirs go =
-    withBuildConfig go $ do
+    withBuildConfigAndLock go $ \_ -> do
```

Comment:
I dunno!  I was being conservative and locking on all commands, except for the carved out exception for `stack exec`/`stack ghci`.  I wasn't totally sure which commands might imply other build actions recursively.

But... `sdistCmd` *is* a clean one..  It's effects seem to be visible in its body without reading over too much other code.   Its effects seem obviously local to the current directory.  Ok, I've made sdist another exception.  For now it instantly unlocks rather than never locking, but that could be improved.

Any more exceptions?